packages/mask/src/plugins/Trader/SNSAdaptor/trader/ConfirmDialog.tsx

```diff
@@ -67,42 +114,161 @@ export function ConfirmDialogUI(props: ConfirmDialogUIProps) {
     }, [trade, executionPrice])
     //#endregion
 
+    //#region gas price
+    const nativeToken = createNativeToken(chainId)
+    const tokenPrice = useNativeTokenPrice(chainId)
+
+    const gasFee = useMemo(() => {
+        return gas && gasPrice ? new BigNumber(gasPrice).multipliedBy(gas).integerValue().toFixed() : 0
+    }, [gas, gasPrice])
+
+    const feeValueUSD = useMemo(
+        () => (gasFee ? new BigNumber(formatWeiToEther(gasFee).times(tokenPrice).toFixed(2).toString()) : '0'),
```

Comment:
Duplicate `toString` and `toFixed`.

```suggestion
        () => (gasFee ? new BigNumber(formatWeiToEther(gasFee).times(tokenPrice).toFixed(2)) : '0'),
```